lib/Controller/PageController.php

```diff
@@ -80,6 +80,27 @@ public function touch(int $collectiveId, int $id): DataResponse {
 		}, $this->logger);
 	}
 
+	public function contentFiltered(int $collectiveId, string $filterString): DataResponse {
+		return $this->handleErrorResponse(function () use ($collectiveId, $filterString): array {
+			$userId = $this->getUserId();
+			$pageInfos = $this->service->findAll($collectiveId, $userId);
+			$contentFilteredPages = [];
+			foreach ($pageInfos as &$pageInfo) {
+				$file = $this->service->getPageFile($collectiveId, $pageInfo->getId(), $userId);
+				$fileContent = $file->getContent();
+				if (str_contains(strtolower($fileContent), strtolower($filterString))) {
+					$contentFilteredPages[] = [
+						"page" => $pageInfo->jsonSerialize(),
+						"content" => $fileContent
+					];
+				}
+			}
```

Comment:
This is a bad idea performance wise as it will read the file content of every file. We had done this in the past but went with a different one.

We actually have a local full text search implementation within collectives that creates an index of the pages which we should rather use here. You can checkout the search provider of the server search for how that is being used: https://github.com/nextcloud/collectives/blob/1c7d1956cee227b2c1fa8df248249d3874927449/lib/Search/PageContentProvider.php#L69

References as we introduced this before moving collectives to github:
https://gitlab.com/collectivecloud/collectives/-/merge_requests/541

While there is also an existing search endpoint that the server provides, I think having our own one here makes sense as we could extend it further with highlighting that TNTsearch also provides.

Reply:
I've implemented it to use the indexed Search. Should we change something abt the indexing? Idk how often it is done, but there are outdated results after writing to a page.